blockly-core/core/ui/block.js

```diff
@@ -1796,7 +1809,7 @@ Blockly.Block.prototype.setNextStatement = function(hasNext, opt_check) {
     this.nextConnection.dispose();
     this.nextConnection = null;
   }
-  if (hasNext) {
+  if (hasNext && !this.nextConnectionDisabled_) {
```

Comment:
oh, very true. Actually, I suppose there's not much point in holding on to that value at all.